control-plane-operator/controllers/hostedcontrolplane/kas/config.go

```diff
@@ -18,11 +18,12 @@ import (
 )
 
 const (
-	KubeAPIServerConfigKey  = "config.json"
-	OauthMetadataConfigKey  = "oauthMetadata.json"
-	AuditLogFile            = "audit.log"
-	EgressSelectorConfigKey = "config.yaml"
-	DefaultEtcdPort         = 2379
+	KubeAPIServerConfigKey    = "config.json"
+	OauthMetadataConfigKey    = "oauthMetadata.json"
+	AuditLogFile              = "audit.log"
+	EgressSelectorConfigKey   = "config.yaml"
+	DefaultEtcdPort           = 2379
+	AuditWebhookKubeconfigKey = "webhook-kubeconfig"
```

Comment:
If you life the key constant out into the public API (which I think is the right thing to do) this solves 1 of the 2 invalid dependencies introduced between hypershift-operator and control-plane-operator below

Reply:
So you want to support them adding any key to the webhook secret?  Currently the assumption is that the secret has the well defined key of `webhook-kubeconfig`. If we are ok with another field in the API that can be done.

Reply:
fixed move constant to public api. I do not believe their is value though in letting the user specify whatever key name they want in the secret to store the kubeconfig since that secret will only contain one key that has the kubeconfig. We can standardize that name with the constant

Reply:
I agree users can't configure the secret internals (e.g. key names) — what I mean is that if there's an API field that is a reference to a secret, I think we should document (on the field itself) what the expected key names and value contents should look like in the secret, and also provide Go constants for the keys. Sort of like [this upstream example](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L5891). We're not doing this uniformly now, it's just something I thought of as we're talking about a new field.

Reply:
done I believe